src/components/views/rooms/RoomTile.js

```diff
@@ -43,43 +44,47 @@ module.exports = React.createClass({
     },
 
     getInitialState: function() {
-        var areNotifsMuted = false;
-        var cli = MatrixClientPeg.get();
-        if (!cli.isGuest()) {
-            var roomPushRule = cli.getRoomPushRule("global", this.props.room.roomId);
-            if (roomPushRule) {
-                if (0 <= roomPushRule.actions.indexOf("dont_notify")) {
-                    areNotifsMuted = true;
-                }
-            }
-        }
-
         return({
             hover : false,
             badgeHover : false,
             notificationTagMenu: false,
             roomTagMenu: false,
-            areNotifsMuted: areNotifsMuted,
+            notifState: this._getNotifState(),
         });
     },
 
-    onAction: function(payload) {
-        switch (payload.action) {
-            case 'notification_change':
-                // Is the notification about this room?
-                if (payload.roomId === this.props.room.roomId) {
-                    this.setState( { areNotifsMuted : payload.areNotifsMuted });
-                }
-                break;
+    _getNotifState: function() {
+        if (MatrixClientPeg.get().isGuest()) return 'all_messages';
+        return RoomNotifs.getRoomNotifsState(this.props.room.roomId);
+    },
+
+    _shouldShowNotifBadge: function() {
+        const showBadgeInStates = ['all_messages', 'all_messages_loud'];
+        const currentState = this._getNotifState();
```

Comment:
why are we calling `_getNotifState` again rather than using `this.state.notifState`?